File: dash_docs/chapters/advanced_callbacks/examples/prevent_update_button.R
library(dash)
library(dashHtmlComponents)
library(dashCoreComponents)

app <- Dash$new()

app$layout(htmlDiv(list(
  htmlButton('Click here to see the content', id='show-secret'),
  htmlDiv(id='body-div')
)))

app$callback(
  output(id='body-div', property='children'),
  params = list(
    input(id='show-secret', property='n_clicks')
  ),
  update_output <- function(n_clicks) {
    if (is.null(n_clicks[[1]])) {
      return(dashNoUpdate())
    }
    else {
      return("Elephants are the only animal that can't jump")
    }
  }
)

app$run_server()

File: dash_docs/chapters/advanced_callbacks/examples/prevent_update_button.py
import dash
import dash_html_components as html
from dash.dependencies import Input, Output
from dash.exceptions import PreventUpdate

external_stylesheets = ['https://codepen.io/chriddyp/pen/bWLwgP.css']

app = dash.Dash(__name__, external_stylesheets=external_stylesheets)

app.layout = html.Div([
    html.Button('Click here to see the content', id='show-secret'),
    html.Div(id='body-div')
])

@app.callback(
    Output(component_id='body-div', component_property='children'),
    [Input(component_id='show-secret', component_property='n_clicks')]
)
def update_output(n_clicks):
    if n_clicks is None:
        raise PreventUpdate
    else:
        return "Elephants are the only animal that can't jump"

if __name__ == '__main__':
    app.run_server(debug=True)
